internal/engine/build_and_deployer_test.go

```diff
@@ -100,233 +100,121 @@ func TestDockerForMacDeploy(t *testing.T) {
-		t.Errorf("Expected no push to docker, actual: %d", f.docker.PushCount)
+	if f.docker.PushCount != 1 {
+		t.Errorf("Expected 1 push to docker, actual: %d", f.docker.PushCount)
 	}
 
 	expectedYaml := "image: gcr.io/some-project-162817/sancho:tilt-11cd0b38bc3ceb95"
 	if !strings.Contains(f.k8s.Yaml, expectedYaml) {
 		t.Errorf("Expected yaml to contain %q. Actual:\n%s", expectedYaml, f.k8s.Yaml)
 	}
 
-	if strings.Contains(f.k8s.Yaml, sidecar.DefaultSyncletImageName) {
-		t.Errorf("Should not deploy the synclet on docker-for-desktop: %s", f.k8s.Yaml)
+	if !strings.Contains(f.k8s.Yaml, sidecar.DefaultSyncletImageName) {
+		t.Errorf("Should deploy the synclet when updateMode=Synclet specified: %s", f.k8s.Yaml)
 	}
 }
 
-func TestSyncletNamespaceGKE(t *testing.T) {
-	f := newBDFixture(t, k8s.EnvGKE, container.RuntimeDocker)
+func TestNoSyncletDeployIfNoLiveUpdate(t *testing.T) {
+	f := newBDFixtureWithUpdateMode(t, k8s.EnvGKE, container.RuntimeDocker, buildcontrol.UpdateModeSynclet)
 	defer f.TearDown()
 
-	assert.Equal(t, "", string(f.sCli.Namespace))
-	assert.Equal(t, "", string(f.k8s.LastPodQueryNamespace))
-
-	manifest := NewSanchoLiveUpdateManifest(f)
+	manifest := NewSanchoDockerBuildManifest(f)
 	targets := buildTargets(manifest)
-	result, err := f.bd.BuildAndDeploy(f.ctx, f.st, targets, store.BuildStateSet{})
+	_, err := f.bd.BuildAndDeploy(f.ctx, f.st, targets, store.BuildStateSet{})
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	cInfo := testContainerInfo
-	cInfo.Namespace = "sancho-ns"
-
-	changed := f.WriteFile("a.txt", "a")
-	bs := resultToStateSet(result, []string{changed}, cInfo)
-	_, err = f.bd.BuildAndDeploy(f.ctx, f.st, buildTargets(manifest), bs)
-	if err != nil {
-		t.Fatal(err)
+	if f.docker.BuildCount != 1 {
+		t.Errorf("Expected 1 docker build, actual: %d", f.docker.BuildCount)
 	}
 
-	assert.Equal(t, "sancho-ns", string(f.sCli.Namespace))
-}
-
-func TestYamlManifestDeploy(t *testing.T) {
-	f := newBDFixture(t, k8s.EnvGKE, container.RuntimeDocker)
-	defer f.TearDown()
+	if f.docker.PushCount != 1 {
+		t.Errorf("Expected 1 push to docker, actual: %d", f.docker.PushCount)
+	}
 
-	manifest := manifestbuilder.New(f, "some_yaml").
-		WithK8sYAML(testyaml.TracerYAML).Build()
-	targets := buildTargets(manifest)
-	_, err := f.bd.BuildAndDeploy(f.ctx, f.st, targets, store.BuildStateSet{})
-	if err != nil {
-		t.Fatal(err)
+	expectedYaml := "image: gcr.io/some-project-162817/sancho:tilt-11cd0b38bc3ceb95"
+	if !strings.Contains(f.k8s.Yaml, expectedYaml) {
+		t.Errorf("Expected yaml to contain %q. Actual:\n%s", expectedYaml, f.k8s.Yaml)
 	}
 
-	assert.Equal(t, 0, f.sCli.UpdateContainerCount)
-	assert.Equal(t, 0, f.docker.BuildCount)
-	assert.Equal(t, 0, f.docker.PushCount)
-	f.assertK8sUpsertCalled(true)
+	if strings.Contains(f.k8s.Yaml, sidecar.DefaultSyncletImageName) {
+		t.Errorf("Should not deploy the synclet if manifest has no LiveUpdate instructions: %s", f.k8s.Yaml)
+	}
 }
 
-func TestContainerBuildLocal(t *testing.T) {
```

Comment:
Deleted a bunch of tests that are covered more concisely in `build_and_deployer_live_update_test.go`